src/features/users/Users.tsx

```diff
@@ -85,7 +88,44 @@ const FilterItemsBlock = styled.div`
 
 const Users: FC = () => {
   const dispatch = useAppDispatch();
-  const users = useAppSelector(selectFilteredUsers);
+  const allUsers = useAppSelector(selectUsers);
+  const sorting = useAppSelector(selectSorting);
+  const searchText = useAppSelector(selectSearchText);
+
+  const sortUsers = (users: User[], sortParams: SortParams) => {
+    switch (sortParams.rule) {
+      case 'ASC':
+        return users.slice().sort((a, b) => {
+          if (a[sortParams.columnName] === b[sortParams.columnName]) return 0;
+          return a[sortParams.columnName] > b[sortParams.columnName] ? 1 : -1;
+        });
+      case 'DSC':
+        return users.slice().sort((a, b) => {
+          if (a[sortParams.columnName] === b[sortParams.columnName]) return 0;
+          return a[sortParams.columnName] < b[sortParams.columnName] ? 1 : -1;
+        });
+      default: return users;
+    }
+  };
```

Comment:
Вот это все можно короче записать, и сделать универсальной функцией, т.к. явных причин привязываться к интерфейсу User здесь нет:

```suggestion
  const sortBy = <T>(list: T[], { rule, columnName }: SortParams<T>) => (
    list.slice().sort((a, b) => (
      rule === 'DESC'
        ? a[columnName] > b[columnName] ? -1 : 1
        : a[columnName] > b[columnName] ? 1 : -1
    ))
  );
```

Ну и интерфейс Sorting нужно будет поправить:

```
export interface Sorting<T> {
  columnName: keyof T;
  rule: 'ASC' | 'DESC';
}
```

Я этот код не проверял, но должно работать.